service/history/nDCActivityReplicator.go

```diff
@@ -215,84 +192,132 @@ func (r *nDCActivityReplicatorImpl) SyncActivity(
 	)
 }
 
-func (r *nDCActivityReplicatorImpl) shouldApplySyncActivity(
+func (r *nDCActivityReplicatorImpl) refreshTask(
+	version int64,
+	attempt int32,
+	activityInfo *persistencespb.ActivityInfo,
+) bool {
+
+	// calculate whether to reset the activity timer task status bits
+	// reset timer task status bits if
+	// 1. same source cluster & attempt changes
+	// 2. different source cluster
+	if !r.clusterMetadata.IsVersionFromSameCluster(version, activityInfo.Version) {
+		return true
+	} else if activityInfo.Attempt != attempt {
+		return true
+	}
+	return false
+}
+
+func (r *nDCActivityReplicatorImpl) testActivity(
+	version int64,
+	attempt int32,
+	lastHeartbeatTime time.Time,
+	activityInfo *persistencespb.ActivityInfo,
+) bool {
+
+	if activityInfo.Version > version {
+		// this should not retry, can be caused by failover or reset
+		return false
+	} else if activityInfo.Version < version {
```

Comment:
maybe convert `if-else if` in sequential if blocks as you are returning from each of those.  Makes code more readable.